### PR TITLE
Fix: remove unnecessary check of nat cases

### PR DESCRIPTION
**Pull request by chloerh**
- Add version check for nat ipv6 test
- Remove ipv6 nat check when not setting nat ipv6 to 'yes'
- Use 'wait_for' for dhcp lease check test

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
```
# avocado run --vt-type libvirt net_nat
JOB ID     : 6e77a0a3e10a7d1a06365695cda00d4e5112d6e2
JOB LOG    : /root/avocado/job-results/job-2020-12-04T18.48-6e77a0a/job.log
 (1/3) type_specific.local.virtual_network.iface_network.net_forward.net_nat.default: PASS (51.41 s)
 (2/3) type_specific.local.virtual_network.iface_network.net_forward.net_nat.nat_ipv6: PASS (54.93 s)
 (3/3) type_specific.local.virtual_network.iface_network.net_forward.net_nat.with_2net: PASS (608.48 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 716.06 s
```

**Comment by chloerh** [comment]
@yalzhang @smitterl @kylazhang Could you help review?

File: libvirt/tests/src/virtual_network/iface_network.py [hunk 1]
```diff
@@ -642,6 +644,10 @@ def run_guest_libvirt(session):
         if not libvirt_version.version_compare(1, 0, 1):
             test.cancel("Not supported Qos options 'floor'")
 
+    if 'nat_ipv6' in params['name']:
+        if not libvirt_version.version_compare(6, 5, 0):
+            test.cancel('Nat ipv6 is not supported untill 6.5.0')
```
Comment (smitterl) [inline libvirt/tests/src/virtual_network/iface_network.py:649]: untill -> until

**Review by yalzhang — APPROVED** [review]

File: libvirt/tests/src/virtual_network/iface_network.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/iface_network.py]: Updated. Thank u.

**Review by smitterl — APPROVED** [review]

**Pull request merged**